rocketpy/mathutils/function.py

```diff
@@ -110,27 +119,25 @@ def __init__(
         (II) Fields in CSV files may be enclosed in double quotes. If fields
         are not quoted, double quotes should not appear inside them.
         """
-
-        inputs, outputs = Function._validate_inputs_outputs(inputs, outputs)
-
         # initialize variables to avoid errors when being called by other methods
-        self.get_value_opt = None
         self.__polynomial_coefficients__ = None
         self.__akima_coefficients__ = None
         self.__spline_coefficients__ = None
-        self.EXTRAPOLATION_TYPES = {
-            "zero": 0,
-            "natural": 1,
-            "constant": 2,
-        }
-
-        # store variables
-        self.set_inputs(inputs)
-        self.set_outputs(outputs)
+
+        # initialize parameters
+        self.source = source
+        self.__inputs__ = inputs
+        self.__outputs__ = outputs
         self.__interpolation__ = interpolation
         self.__extrapolation__ = extrapolation
-        self.set_source(source)
-        self.set_title(title)
+        self.title = title
+        self.__img_dim__ = 1  # always 1, here for backwards compatibility
+
+        # args must be passed from self.
```

Comment:
why?

Reply:
The `_check_user_input` had a check to guarantee that `len(outputs)==1`. So this would always end up being one.